app/mobile/lib/view/resource/resourcepage.dart

```diff
@@ -42,16 +49,70 @@ Widget resourcePageView(TopicDetailed t, Resource r, ChangeNotifier topicPageVie
                               }
                             },
                             styleSheet: MarkdownStyleSheet(
-                                blockquoteDecoration:
-                                    const BoxDecoration(color: CustomColors.main)),
+                                blockquoteDecoration: const BoxDecoration(
+                                    color: CustomColors.main)),
                             data: r.body,
                             shrinkWrap: true,
                             physics: const ClampingScrollPhysics(),
                           ),
                         ],
                       ),
                     ),
-                    editResourceButton(t, r, context, topicPageView),
+                    Padding(
+                      padding: const EdgeInsets.all(10.0),
+                      child: AnimatedFloatingActionButton(
+                          fabButtons: <Widget>[
+                            FloatingActionButton(
+                              onPressed: () => editResourceButton(
+                                  t, r, context, topicPageView),
+                              backgroundColor: CustomColors.main,
+                              tooltip: 'Edit Resource',
+                              heroTag: 'btn1',
+                              child: const Icon(
+                                Icons.edit_outlined,
+                              ),
+                            ),
+                            FloatingActionButton(
+                              onPressed: null,
+                              backgroundColor: CustomColors.main,
+                              tooltip: 'Create A New Note',
+                              heroTag: 'btn2',
+                              child: Icon(Icons.note_add_outlined),
+                            ),
+                            FloatingActionButton(
+                              onPressed: () async => {
+                                if (r.discussion?.id != null)
+                                  {
+                                    PersistentNavBarNavigator.pushNewScreen(
+                                        context,
+                                        screen: discussionView(
+                                            discussionId: r.discussion!.id))
+                                  }
+                                else //TODO: This part may be wrong needs further checking
```

Comment:
Instead of having the same PersistentNavBarNavigator call twice, we can check for
```dart
if(d.discussion?.id == null) {// get discussion from service}
// PersistentNavBarNavigator call

Reply:
I changed this part to fit your description. But I am not very sure I got u right please check.